UserInputValidator.cs

```diff
@@ -0,0 +1,27 @@
+namespace TictactoeVer2
+{
+    public class UserInputValidator
+    {
+        public bool IsBoardSizeValid(string input)
+        {
+            var size = ParseBoardSize(input);
```

Comment:
Consider the `TryParseBoardSize(string input, out int size)` approach at this level, to avoid the implicit meaning of -1.
`ParseBoardSize` as a method name does not describe any failure condition, or even the possibility of failure.